### ISSUE TITLE
Autocomplete for xepersian commands?

### ISSUE THREAD

**Issue by rezass** [latex-lsp/texlab#834]
Hi all and happy new year!

Is there support for commands provided by xepersian package? Commands defined like the ones in Page 7 of http://mirrors.ibiblio.org/CTAN/macros/xetex/latex/xepersian/xepersian-doc.pdf. If not, can you point me to the right direction to include them? [components.json.gz](https://github.com/latex-lsp/texlab/blob/master/data/components.json.gz) seems to be the correct place, but can I just add unicode strings?

Best wishes,
Reza

**Comment by pfoerster** [comment]
@rezass

> If not, can you point me to the right direction to include them?

Sure, the `components.json.gz` is auto-generated with a Python script from the https://github.com/latex-lsp/latex-completion-data repository. Most of the packages are scanned automatically by basically asking the TeX engine if a particular command exists for a specific package. The candidates to check are extracted from the source code. For `xepersian`, the scripts find a few commands but not all of them (especially the RTL variants are missing). 
In cases where the package cannot be scanned automatically, we a special file ([`appendix.json`](https://github.com/latex-lsp/latex-completion-data/blob/master/latex-completion-data/data/appendix.json)), that is merged into the final file.

To summarize, I think it is easiest to include the missing commands in the `appendix.json` file and re-run the script. The script itself might need to be fixed somewhere since I did not update the cache for a while.

**Comment by rezass** [comment]
Thanks @pfoerster 

I checked that repo and will add the RTL commands in the appendix. Will you accept pull requests for RTL compatibility both in that repo and texlab? I want to also enable RTL equivalent of section, chapter, ... in texlab. The changes would be  like adding the below code:

```
#[token("\\section")]
#[token("\\بخش")]
#[token("\\section*")]
#[token("\\بخش*")]
Section,
```

**Comment by pfoerster** [comment]
@rezass Sure, this looks easy enough to maintain 👍 . Thanks for contributing!

**Comment by rezass** [comment]
Perfect! So I will send my pull requests.
